From 3d3ee2f91bbcb364891e82b99e932d918c605658 Mon Sep 17 00:00:00 2001
From: iain holmes <iain@xamarin.com>
Date: Thu, 24 Nov 2016 11:30:23 +0000
Subject: [PATCH] [A11y] Fix dialog accessibility

Composite widgets like the dialog need to have their accessibility element
created before they start building themselves.
---
 gtk/gtkdialog.c | 8 ++++++++
 1 file changed, 8 insertions(+)

diff --git a/gtk/gtkdialog.c b/gtk/gtkdialog.c
index a2756f6..41fc9b1 100644
--- a/gtk/gtkdialog.c
+++ b/gtk/gtkdialog.c
@@ -275,6 +275,14 @@ gtk_dialog_init (GtkDialog *dialog)
   priv = GET_PRIVATE (dialog);
   priv->ignore_separator = FALSE;
 
+  /* If we don't create the accessible now, then the accessibility subsystem
+   * will never know about vbox being added. This is a bit of a hack as
+   * every composite widget that builds itself in _init will need to do this
+   * or else it will be inaccessible, but I can't think of a better way at
+   * present
+   */
+  gtk_widget_get_accessible (GTK_WIDGET (dialog));
+
   /* To avoid breaking old code that prevents destroy on delete event
    * by connecting a handler, we have to have the FIRST signal
    * connection on the dialog.
-- 
2.9.3 (Apple Git-75)

